### ISSUE TITLE
Mysterious ticking noise at each turn

### ISSUE THREAD

**Issue by akien-mga** [OpenDungeons/OpenDungeons#52]
While in game, I hear a ticking noise at each passing turn, as if a timer was running.

@tomluchowski told me he heard it too, but I could not understand whether it was a bug or a feature (in which case it should be made clearer what this ticking noise means, because it sounds like a sound handling bug).

**Comment by Bertram25** [comment]
@akien-mageia That "ticking" sound comes from the fact that some worker is digging somewhere.
The digging sound is triggered each turn, explaining why it is sounding like some ticking:

https://github.com/OpenDungeons/OpenDungeons/blob/development/source/Creature.cpp#L1471

More precisely, the digging sound triggered is handled with SFML as a spatial with a position, a velocity, ... but the values used are wrong as the sound should fade out much more quickly when the camera go away from them.

The actual fix would then IMHO to setup the strength of the sound more correctly and make some tests.

**Comment by Bertram25** [comment]
Can be set to later, as it is not blocking and wasn't identified as urgent before release.

**Comment by akien-mga** [comment]
That's true, though I wouldn't mind if we could use another sound in the meantime that does not sound like the ticking noise of a clock (especially since it's so regular and brief). Maybe something more "rocky" that sounds like pickaxe on rocks? @Danimal696 WDYT?

**Comment by Bertram25** [comment]
@akien-mga Sure, just check the Digging available sounds and have a go changing the name here:
https://github.com/OpenDungeons/OpenDungeons/blob/development/source/SoundEffectsHelper.cpp#L113

**Comment by akien-mga** [comment]
I'd like to set this one back to the 0.4.9, I think there is room for easy improvement. Even if we can't get the spatial intensity, we could probably make it so that the sound played at each tick is chosen randomly among the available sounds in sounds/Digging. We have 18 of them, and they add a great deal of variety to the digging tick :-)

@hwoarangmy @Bertram25 WDYT?

**Comment by Bertram25** [comment]
Alright, I'll take that one and will deal with it, once the main menu issue is done. :)

**Comment by Danimal696** [comment]
i will try to look for new ones, cant promise anything; im having the final rush for a recovery test. Will probably have to wait until i finish that.

**Comment by akien-mga** [comment]
@Danimal696 Actually the current ones are quite good in my opinion, I hadn't listened to them when I asked you for replacements. The problem is just that only one of them is used for now (and that the sound should only be played when looking at workers).

**Comment by Danimal696** [comment]
it will be the same for creature step, cries, death... sounds; it would be nice if someone fix it in preparation for that.

**Comment by Bertram25** [comment]
I'm on it. From what I was able to code so far, creatures will be able to play a random sound for each action type. I also separated more clearly the interface sounds from the creature ones.

**Comment by Bertram25** [comment]
For the record, once my pull request will be fixed and merged in, the game will use every available sounds for a given action in the corresponding folders and play them at random and with a revised spatial position, tested by @akien-mga ;)

This also means you can add new sounds in a sounds/ subfolder and they will be taken in account. E.g:
If you add a new sound in sounds/Digging/, it will be used at random when digging on the next game restart.

Also, before we add creature idle sound support, I'd rather fix the naive parts of my code first (in another pull request, and probably after the release.), and add actual configurable sound per creature class support, too.
But again, I'll be honest, I'd rather improve the music handling first. ;)

**Comment by Danimal696** [comment]
its already a pretty good patch! nice work bertram

**Comment by akien-mga** [comment]
Fixed by #118.
